cmd/list.go

```diff
@@ -36,34 +38,59 @@ var listCmd = &cobra.Command{
 	RunE:  runList,
 }
 
-func roleList(all bool) (string, error) {
+func roleList() (string, error) {
 	client, err := creds.GetClient(region)
 	if err != nil {
 		return "", err
 	}
-	roles, err := client.Roles()
+	roles, err := client.RolesExtended()
 	if err != nil {
 		return "", err
 	}
-	var sb strings.Builder
-	if all {
-		sb.WriteString("Available Roles\n")
-	} else {
-		sb.WriteString("Available Console Roles\n")
+	var rolesData [][]string
+	for _, role := range roles {
+		if shortInfo {
+			rolesData = append(rolesData, []string{role.Arn})
+			continue
+		}
+		curData := []string{role.AccountName, role.RoleName, role.Arn}
+		if extendedInfo {
+			var namesb strings.Builder
+			var ownersb strings.Builder
+			for _, app := range role.Apps.AppDetails {
+				namesb.WriteString(app.Name)
+				namesb.WriteString("\n")
+				ownersb.WriteString(app.Owner)
+				ownersb.WriteString("\n")
+			}
+			appNames := namesb.String()
+			ownerNames := ownersb.String()
+			if len(appNames) > 0 {
+				curData = append(curData, appNames[:len(appNames)-1])
+				curData = append(curData, ownerNames[:len(ownerNames)-1])
+			}
+		}
+		rolesData = append(rolesData, curData)
 	}
-	for i := range roles {
-		sb.WriteString(roles[i])
-		sb.WriteString("\n")
+	var headers []string
+	if shortInfo {
```

Comment:
This is pretty nifty. I like that we can use the same input data and change the output based on the headers we pass in. 👍